pkg/webhooks/server.go

```diff
@@ -18,9 +19,104 @@ import (
 )
 
+type AdmissionRequestPayload struct {
+	UID                types.UID                    `json:"uid"`
+	Kind               metav1.GroupVersionKind      `json:"kind"`
+	Resource           metav1.GroupVersionResource  `json:"resource"`
+	SubResource        string                       `json:"subResource,omitempty"`
+	RequestKind        *metav1.GroupVersionKind     `json:"requestKind,omitempty"`
+	RequestResource    *metav1.GroupVersionResource `json:"requestResource,omitempty"`
+	RequestSubResource string                       `json:"requestSubResource,omitempty"`
+	Name               string                       `json:"name,omitempty"`
+	Namespace          string                       `json:"namespace,omitempty"`
+	Operation          string                       `json:"operation"`
+	UserInfo           authenticationv1.UserInfo    `json:"userInfo"`
+	Object             map[string]interface{}       `json:"object,omitempty"`
+	OldObject          map[string]interface{}       `json:"oldObject,omitempty"`
+	DryRun             *bool                        `json:"dryRun,omitempty"`
+	// Options runtime.RawExtension `json:"options,omitempty" protobuf:"bytes,12,opt,name=options"`
+}
+
+func newAdmissionRequestPayload(rq *admissionv1.AdmissionRequest) (*AdmissionRequestPayload, error) {
+	newResource, oldResource, err := utils.ExtractResources(nil, rq)
+	if err != nil {
+		return nil, err
+	}
+	return &AdmissionRequestPayload{
+		UID:                rq.UID,
+		Kind:               rq.Kind,
+		Resource:           rq.Resource,
+		SubResource:        rq.SubResource,
+		RequestKind:        rq.RequestKind,
+		RequestResource:    rq.RequestResource,
+		RequestSubResource: rq.RequestSubResource,
+		Name:               rq.Name,
+		Namespace:          rq.Namespace,
+		Operation:          string(rq.Operation),
+		UserInfo:           rq.UserInfo,
+		Object:             newResource.Object,
+		OldObject:          oldResource.Object,
+		DryRun:             rq.DryRun,
+	}, nil
+}
+
+var (
+	dumpPayload bool
+	// dumpPayloadTimer time.Duration
+	dumpPayloadOps   []string
+	dumpPayloadKinds []string
+)
+
+func splitString(s string) []string {
+	return strings.Split(s, " ")
+}
+
+func dumpOperation(op string) bool {
+	for _, o := range dumpPayloadOps {
+		if strings.EqualFold(o, op) {
+			return true
+		}
+	}
+	return false
+}
+
+func dumpKind(kind string) bool {
+	for _, k := range dumpPayloadKinds {
+		if k == kind {
+			return true
+		}
+	}
+	return false
+}
+
+func SetDumpFlags(dumpPl bool, dumpPlOpts ...string) {
+	dumpPayload = dumpPl
+	// duration := dumpPlOpts[0][:len(dumpPlOpts[0])-1]
+	// dumpPayloadTimer = time.Second * time.Duration(60)
+	dumpPayloadOps = splitString(dumpPlOpts[1])
+	dumpPayloadKinds = splitString(dumpPlOpts[2])
+}
+
+func redactSecretKind(payload *AdmissionRequestPayload) {
+	obj, oldObj := payload.Object, payload.OldObject
+	if obj != nil {
+		delete(obj, "data")
+		x := obj["metadata"].(map[string]interface{})
+		delete(x, "annotations")
```

Comment:
You have helpers for that

Reply:
you mean helpers that are already in the codebase? or defining myself?

Reply:
@eddycharly what are these helpers?

Reply:
In unstructured package, see an example here https://github.com/kyverno/kyverno/blob/5a09a783505642c36d4df130327194e99cab259b/pkg/utils/report/labels.go#L80